torch/optim/optimizer.py

```diff
@@ -326,15 +421,18 @@ def register_step_pre_hook(self, hook: Callable[..., None]) -> RemovableHandle:
             hook (Callable): The user defined hook to be registered.
 
         Returns:
-            :class:`torch.utils.hooks.RemoveableHandle`:
+            :class:`torch.utils.hooks.RemovableHandle`:
                 a handle that can be used to remove the added hook by calling
                 ``handle.remove()``
         """
         handle = hooks.RemovableHandle(self._optimizer_step_pre_hooks)
         self._optimizer_step_pre_hooks[handle.id] = hook
         return handle
 
-    def register_step_post_hook(self, hook: Callable[..., None]) -> RemovableHandle:
+    def register_step_post_hook(
+        self,
+        hook: Callable[[Self, Tuple[Any, ...], Dict[str, Any]], None],  # type: ignore[valid-type]
```

Comment:
Why not use the type alias OptimizerPostHook?

Reply:
addressed with class variable `TypeAlias`